### PR TITLE
Path comparison updates

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

At some point these two binaries got out of date.  Updated.  

Sample command
`./valhalla_path_comparison -s 'crjgkA~lthpCLiMxBofBgm@{AgM?uZxCU?' -t auto --config ../../conf/valhalla.conf`

```
+++++++++++++++++++++++++++++++++++++
wayid: 11929406
name: West Main Street
+++++++++++++++++++++++++++++++++++++

----------Edge----------
Edge GraphId: 0/2905/191349
Edge length: 130
EdgeCost cost: 10.8225 secs: 11.7
------------------------

-------Transition-------
Pred GraphId: 0/2905/191349
TransitionCost cost: 0 secs: 0
------------------------

----------Edge----------
Edge GraphId: 0/2905/192090
Edge length: 141
EdgeCost cost: 11.7383 secs: 12.69
------------------------

+++++++++++++++++++++++++++++++++++++
wayid: 76951972
name: North Decatur Street
+++++++++++++++++++++++++++++++++++++

-------Transition-------
Pred GraphId: 0/2905/192090
TransitionCost cost: 7.5 secs: 7.5
------------------------

----------Edge----------
Edge GraphId: 0/2905/191361
Edge length: 202
EdgeCost cost: 16.8165 secs: 18.18
------------------------

+------------------------------------------------------------------------+
| Total Edge Cost       :    39.3773  Total Edge Secs       :      42.57 |
| Total Transition Cost :        7.5  Total Transition Secs :        7.5 |
| Total Cost            :    46.8773  Total Secs            :      50.07 |
+------------------------------------------------------------------------+

```

## Tasklist

 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by danpat** [comment]
@gknisely can you add a test for these so we don't get future similar regressions?

File: src/valhalla_run_route.cc [hunk 1]
```diff
@@ -206,6 +206,20 @@ const valhalla::TripLeg* PathTest(GraphReader& reader,
     const auto projections = Search(locations, reader, cost);
     std::vector<PathLocation> path_location;
     valhalla::Options options;
+
+    for (const auto& ll : shape) {
```
Comment (kevinkreiser) [inline src/valhalla_run_route.cc:210]: shape is only needed for map matching. regular routing doesnt use it. is this used for map matching?

Comment (gknisely) [reply src/valhalla_run_route.cc:210]: yes.  https://github.com/valhalla/valhalla/blob/master/src/valhalla_run_route.cc#L214

**Comment by gknisely** [comment]
> @gknisely can you add a test for these so we don't get future similar regressions?

@kevinkreiser is going to add an issue to rewrite these programs to use the actor interface.  We can do it then.

**Pull request merged**